src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -314,9 +311,36 @@ public void switchIndex(final int i, final int j)
     }
 
     @Override
-    public boolean canRecipeBeAdded(final IToken<?> ignored)
+    public boolean canRecipeBeAdded(final IToken<?> token)
     {
-        return hasSpaceForMoreRecipes();
+        return hasSpaceForMoreRecipes() && isRecipeCompatibleWithCraftingModule(token);
+    }
+
+    /**
+     * @param token the recipe token
+     * @return whether the recipe can be added according to the crafting module (or false if there's no module)
+     */
+    protected boolean isRecipeCompatibleWithCraftingModule(final IToken<?> token)
+    {
+        return this.getModules(ICraftingBuildingModule.class).stream()
```

Comment:
I have a feeling that a simple for each loop here could be much easier and much more efficient.